src/main/java/teammates/common/util/ActivityLogEntry.java

```diff
@@ -1,365 +1,308 @@
+    public long getLogTime() {
+        return logTime;
     }
 
-    public boolean isMasqueradeUserRole() {
-        return role.contains(Const.ActivityLog.ROLE_MASQUERADE_POSTFIX);
+    public long getActionTimeTaken() {
+        return actionTimeTaken;
     }
 
-    public String getRole() {
-        return role;
+    public String getActionName() {
+        return actionName;
     }
 
-    public String getName() {
-        return name;
+    public String getActionResponse() {
+        return actionResponse;
     }
 
-    public String getGoogleId() {
-        return googleId;
+    public String getUserRole() {
+        return userRole;
     }
 
-    public String getEmail() {
-        return email;
+    public boolean isMasqueradeUserRole() {
+        return isMasqueradeUserRole;
     }
 
-    public long getTimeTaken() {
-        return timeTaken;
+    public String getUserName() {
+        return userName;
     }
 
-    public String getUrl() {
-        return url;
+    public String getUserGoogleId() {
+        return userGoogleId;
     }
 
-    public String getMessage() {
-        return message;
+    public String getUserEmail() {
+        return userEmail;
     }
 
     /**
-     * Generates the ID for the log. If the googleId is unknown or unregistered,
-     * the email and course of the {@code student} will be used to construct the id.
-     * @param googleId the google ID
-     * @param student StudentAttributes object
-     * @return log ID
+     * Checks whether the activity log is related to testing data or not.
      */
-    public String generateLogId(String googleId, StudentAttributes student, long time) {
-        return student == null ? generateLogId(googleId, null, null, time)
-                               : generateLogId(googleId, student.email, student.course, time);
+    public boolean isTestingData() {
+        return userEmail.endsWith(Const.ActivityLog.TESTING_DATA_EMAIL_POSTFIX);
     }
 
     /**
-     * Generates the ID for the log. If the googleId is unknown or unregistered,
-     * the {@code email} and {@code course} will be used to construct the id.
-     * @param googleId the google ID
-     * @param email the email
-     * @param course the course
-     * @return log ID
+     * Constructs an {@link ActivityLogEntry} from {@link AppLogLine} provided by GAE.
+     *
+     * <p>If the log message in {@link AppLogLine} is not in desired format, an instance will still
+     * be constructed, but with log message: {@link Const.ActivityLog.MESSAGE_ERROR_LOG_MESSAGE_FORMAT}
      */
-    public String generateLogId(String googleId, String email, String course, long time) {
-        boolean isUnregisteredStudent = (googleId.contentEquals("Unknown") || googleId.contentEquals("Unregistered"))
-                                        && email != null && course != null;
+    public static ActivityLogEntry constructFromAppLog(AppLogLine appLog) {
```

Comment:
Perhaps this should be moved to the Builder, because Builder should be the one doing all the constructing? It will eliminate the two-way coupling between the two classes.

BTW, we can use `build` instead of `construct` for consistency.